build_tools/python/e2e_test_framework/definitions/common_definitions.py

```diff
@@ -47,8 +52,8 @@ class DeviceArchitecture(_ArchitectureInfo, Enum):
   RV32_GENERIC = (ArchitectureType.CPU, "riscv_32", "generic")
 
   # Mobile GPUs
-  VALHALL_MALI = (ArchitectureType.GPU, "valhall", "mali")
-  ADRENO_GENERIC = (ArchitectureType.GPU, "adreno", "generic")
+  ADRENO_GENERIC = (ArchitectureType.GPU, "adreno", "", "qualcomm")
+  MALI_VALHALL = (ArchitectureType.GPU, "", "valhall", "arm")
```

Comment:
If ARM is the vendor of GPU, should this be:
```suggestion
  MALI_VALHALL = (ArchitectureType.GPU, "valhall", "", "arm")
```

Reply:
Yeah, this works too, given there is a fuzzy line between the ISA and microarchitecture in various GPU vendors.. Done.